packages/transaction-controller/src/utils/simulation-api.ts

```diff
@@ -189,13 +191,33 @@ export async function simulateTransactions(
  * @param chainId - The chain ID to get the URL for.
  * @returns The URL for the transaction simulation API.
  */
-function getUrl(chainId: Hex): string {
-  const subdomain = SUBDOMAIN_BY_CHAIN_ID[chainId];
+async function getSimulationUrl(chainId: Hex): Promise<string> {
+  const networkData = await getNetworkData();
+  const chainIdDecimal = convertHexToDecimal(chainId);
+  const network = networkData[chainIdDecimal];
 
-  if (!subdomain) {
+  if (!network?.confirmations) {
     log('Chain is not supported', chainId);
     throw new Error(`Chain is not supported: ${chainId}`);
   }
 
+  return getUrl(network.network);
+}
+
+/**
+ * Retrieve the supported network data from the simulation API.
+ */
+async function getNetworkData(): Promise<SimulationNetworkResponse> {
+  const url = `${getUrl('ethereum-mainnet')}${ENDPOINT_NETWORKS}`;
```

Comment:
Nit: You could put the ENDPOINT_NETWORKS suffix in getUrl, since it is part of the url. Or you could just inline getUrl right here since it's so simple now.

Reply:
But I'm also using `getUrl` above in `getSimulationUrl` where I wouldn't want to include the endpoints.